pkg/reconciler/revision/resources/deploy.go

```diff
@@ -370,6 +370,24 @@ func MakeDeployment(rev *v1.Revision, cfg *config.Config) (*appsv1.Deployment, e
 		progressDeadline = int32(pd.Seconds())
 	}
 
+	startupProbeMaxDuration := int32(0)
+	for _, container := range podSpec.Containers {
+		if container.StartupProbe != nil {
+			maxSuccessDuration := container.StartupProbe.PeriodSeconds *
+				container.StartupProbe.SuccessThreshold *
+				container.StartupProbe.TimeoutSeconds
+
+			maxFailDuration := container.StartupProbe.PeriodSeconds *
+				container.StartupProbe.FailureThreshold *
+				container.StartupProbe.TimeoutSeconds
+
+			maxDuration := max(maxSuccessDuration, maxFailDuration)
+			if maxDuration > startupProbeMaxDuration {
+				startupProbeMaxDuration = container.StartupProbe.InitialDelaySeconds + maxDuration
+			}
+		}
+	}
+
```

Comment:
I'm sorta inclined to think that the progressdeadline encapsulates all the (startup time + ready time).

So I'm thinking of not adding this extra time here just because there's a startup probe.

Reply:
Probably we can omit that. It seems that the only validation that happens at the K8s side for progressdeadline is about `MinReadySeconds`, https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/apps/validation/validation.go#L592. I am wondering how that field affects our stuff. :thinking:

Reply:
Hm I think we can, just means a user has to be aware of it and set that field himself accordingly. As startup probes are potentially used for slow-starting things the increase is quite necessary. Otherwise we will never see the pod starting up. Im fine with both ways.

Reply:
@dprotaso @skonto I removed the increase. I'll add it to the docs. PTAL.